### DISCUSSION TITLE
Pre-define y axis for plots

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#1687] [Q&A]
When I plot some columns, I want the y min and max to be set in advance. i.e. for metrics that could go between 0 and 1, the y axis should go between 0 and 1.

I know I'll need to customize VisiData for that, but what would be a good path for that?

Basically, how can I tell VisiData "whenever I ask to plot a column that matches this regex, I'd like you to set to y min and max to 0 and 1 respectively?"

**Comment by saulpw** [comment]
It sounds like you want the general ability to run arbitrary commands or functions after a load finishes.  That's tricky but it's definitely on my radar.

In the meantime, you can set the x and y axis bounds with the `x` and `y` commands on the GraphSheet.  You should be able to turn these into a macro or custom command for your use case fairly easily.

**Reply by cool-RR** [reply]
I understand, but I want the graph to load with my predetermined y axis bounds on the first time. Could I add columns to the column sheet like `default_y_axis_min` and `default_y_axis_max`, and the plot command would read them before deciding on the y axis?

**Reply by saulpw** [reply]
You could have a column `y_axis_bounds` and have both the min and the max in there, and also plot that column.  Otherwise this would be an option for the graph, but again I think the more general solution is to provide you with the ability to run arbitrary commands after loading finishes.

**Reply by cool-RR** [reply]
I ended up just expanding my `autotune` command to call this in case of a graph sheet:

```
@GraphSheet.api
def guess_y_axis(graph_sheet: GraphSheet) -> None:
    values = tuple(
        column.getValue(row) for column in graph_sheet.ycols for row in column.sheet.rows
    )
    min_value = min(values)
    max_value = max(values)
    if 0 <= min_value <= max_value <= 1:
        graph_sheet.set_y('0 1') 
```

I think this'll work good enough for me.
